miden-crypto/src/dsa/ecdsa_secp256k1/mod.rs

```diff
@@ -0,0 +1,397 @@
+//! ECDSA (Elliptic Curve Digital Signature Algorithm) signature implementation over secp256k1
+//! curve.
+
+use alloc::{string::ToString, vec::Vec};
+
+use k256::{
+    ecdh::diffie_hellman,
+    ecdsa::{RecoveryId, SigningKey, VerifyingKey, signature::hazmat::PrehashVerifier},
+};
+use rand::{CryptoRng, RngCore};
+use thiserror::Error;
+
+use crate::{
+    Felt, SequentialCommit, StarkField, Word,
+    ecdh::{EphemeralPublicKey, SharedSecret},
+    utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable},
+};
+
+#[cfg(test)]
+mod tests;
+
+// CONSTANTS
+// ================================================================================================
+
+/// Length of secret key in bytes
+const SECRET_KEY_BYTES: usize = 32;
+/// Length of public key in bytes when using compressed format encoding
+pub(crate) const PUBLIC_KEY_BYTES: usize = 33;
+/// Length of signature in bytes using our custom serialization
+const SIGNATURE_BYTES: usize = 66;
+/// Length of signature in bytes using standard serialization i.e., `SEC1`
+const SIGNATURE_STANDARD_BYTES: usize = 64;
+/// Length of scalars for the `secp256k1` curve
+const SCALARS_SIZE_BYTES: usize = 32;
+
+// ECDSA HASHER
+// ================================================================================================
+
+#[repr(u8)]
+#[derive(Debug, Clone, Copy)]
+pub enum EcdsaHasher {
+    Sha256 = 0,
+    Keccak = 1,
+}
```

Comment:
As discussed in recent comments, I think we should get rid of this for now and always use `Keccak` for hashing.